projects/go-lib/src/lib/components/go-button/go-button.component.spec.ts

```diff
@@ -24,4 +24,145 @@ describe('GoButtonComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  describe('clicked()', () => {
+    it('sets isProcessing to false if useLoader is false', () => {
+      expect(component.isProcessing).toBe(false);
+      component.clicked();
+      expect(component.isProcessing).toBe(false);
+    });
+
+    it('sets isProcessing to true if useLoader is true', () => {
+      component.useLoader =  true;
+
+      expect(component.isProcessing).toBe(false);
+      component.clicked();
+      expect(component.isProcessing).toBe(true);
+    });
+
+    it('emits the handleClick event with the isProcessing boolean', () => {
+      spyOn(component.handleClick, 'emit');
+      spyOn(component, 'clicked').and.callThrough();
+
+      const goButtonTemplate: HTMLElement = fixture.nativeElement;
+      const buttonElement: HTMLButtonElement = goButtonTemplate.querySelector('button');
+
+      buttonElement.dispatchEvent(new Event('click'));
+      fixture.detectChanges();
+
+      expect(component.clicked).toHaveBeenCalled();
+      expect(component.handleClick.emit).toHaveBeenCalledWith(component.isProcessing);
+    });
+  });
+
+  describe('reset()', () => {
+    it('sets isProcessing to false', () => {
+      component.isProcessing = true;
+
+      expect(component.isProcessing).toBe(true);
+      component.reset();
+      expect(component.isProcessing).toBe(false);
+    });
+  });
+
+  describe('classObject()', () => {
+    it('builds an object with all of the class values falsey', () => {
+      const classes: object = component.classObject();
+
+      expect(Object.values(classes)).not.toContain(true);
+    });
+
+    it('returns an object that sets go-button--negative to true based on buttonVariant', () => {
+      expect(component.classObject()['go-button--negative']).toBe(false);
+
+      component.buttonVariant = 'alert';
+      expect(component.classObject()['go-button--negative']).toBe(true);
+
+      component.buttonVariant = 'negative';
+      expect(component.classObject()['go-button--negative']).toBe(true);
+    });
+
+    it('returns an object that set go-button--neutral to true based on buttonVariant', () => {
+      expect(component.classObject()['go-button--neutral']).toBe(false);
+
+      component.buttonVariant = 'neutral';
+      expect(component.classObject()['go-button--neutral']).toBe(true);
+    });
+
+    it('returns an object that set go-button--positive to true based on buttonVariant', () => {
+      expect(component.classObject()['go-button--positive']).toBe(false);
+
+      component.buttonVariant = 'positive';
+      expect(component.classObject()['go-button--positive']).toBe(true);
+    });
+
+    it('returns an object that set other modifiers to true', () => {
```

Comment:
Should these be separated out into individual tests?

Reply:
Meh, I didn't really want to have a bunch of tests to test individual keys in the object. We could separate them in the future if we would like.